### PR TITLE
UltradnsProvider: Add support for ALIAS type (fix #644)

### LINKED ISSUE #644: Support UltraDNS's Apex Alias records
It seems that currently `octodns` does not support the Apex Alias records that one can configure on UltraDNS.
E.g. trying to dump a zone from UltraDNS with such records causes `octodns-dump` to crash:

```
octodns-dump --config-file ./config/staging.yaml --output-dir ./ stg.XXX.com. ultradns                                                                                                                   
2020-12-10T16:27:50  [4492312000] INFO  Manager __init__: config_file=./config/staging.yaml
2020-12-10T16:27:50  [4492312000] INFO  Manager __init__:   max_workers=2
2020-12-10T16:27:50  [4492312000] INFO  Manager __init__:   include_meta=False
2020-12-10T16:27:51  [4492312000] INFO  Manager dump: zone=stg.XXX.com., sources=()
Traceback (most recent call last):
  File "/Users/ndenev/te/octodns/venv/bin/octodns-dump", line 8, in <module>
    sys.exit(main())
  File "/Users/ndenev/te/octodns/venv/lib/python3.9/site-packages/octodns/cmds/dump.py", line 34, in main
    manager.dump(args.zone, args.output_dir, args.lenient, args.split,
  File "/Users/ndenev/te/octodns/venv/lib/python3.9/site-packages/octodns/manager.py", line 469, in dump
    source.populate(zone, lenient=lenient)
  File "/Users/ndenev/te/octodns/venv/lib/python3.9/site-packages/octodns/provider/ultra.py", line 290, in populate
    _type = self.RECORDS_TO_TYPE[record['rrtype']]
KeyError: 'APEXALIAS (65282)'
```

### PR DESCRIPTION
**Pull request by sbrinkerhoff**
- Update API documentation to publicly available URL
- Test and update the currently observed UltraDNS version
- Add support for the Octodns ALIAS type (APEXALIAS in UltraDNS parlance)
- Update zone query limit from 100 to 1000

### CODE REVIEW

File: octodns/provider/ultra.py
Comment (ross) [inline octodns/provider/ultra.py:462]: Best not to change the `Record` type here .  I believe the following should be sufficient. 

```suggestion
                existing_type = existing._type
                if existing_type == "ALIAS":
                    existing_type = "APEXALIAS"

                path = '/v2/zones/{}/rrsets/{}/{    }'.format(zone_name,
                                                          existing_type,
```

File: octodns/provider/ultra.py
Comment (ross) [inline octodns/provider/ultra.py:462]:
```suggestion
                path = '/v2/zones/{}/rrsets/{}/{}'.format(zone_name,
```

**Comment by ross** [comment]
Fixed a whitespace issue, but looks like the tests are failing to find `_data_for_ALIAS`.

**Comment by sbrinkerhoff** [comment]
@ross Cleaned up the unittests/linting/incomplete code provider.  The tests should all pass now (they do locally) with 100% coverage.

**Review by ross — APPROVED** [review]
Looks good. I think I had to re-approve CI runs b/c the last two commits were done with a different "author" from the original bits, otherwise it would have run for you after the first round of approvals. 

Screen Shot 2021-05-28 at 7 13 54 AM

**Pull request merged**
